src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,220 @@
+import './styles.css';
+
+/**
+ * A multi-panel view that the user can add any number of 'panels'.
+ * 'a panel' consists of two elements. Even index element becomes heading,
+ * and odd index element becomes the expandable content.
+ */
+
+function getClosestHeading(el: Element) {
+  const closestEl = el.closest('multi-panel > *');
+  if (closestEl && closestEl.classList.contains('panel-heading')) {
+    return closestEl;
+  }
+  return undefined;
+}
+
+export default class MultiPanel extends HTMLElement {
+
+  constructor() {
+    super();
+
+    // add EventListeners
+    this.addEventListener('click', this._onClick);
+    this.addEventListener('keydown', this._onKeyDown);
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
+  }
+
+  connectedCallback() {
+    this._childrenChange();
+  }
+
+  // Click event handler
+  private _onClick(event: MouseEvent) {
+    const el: Element = event.target as Element;
+    const heading = getClosestHeading(el);
+    if (!heading) return;
+    this._expand(heading);
+  }
+
+  // KeyDown event handler
+  private _onKeyDown(event: KeyboardEvent) {
+    const selectedEl = document.activeElement;
+    const heading = getClosestHeading(selectedEl);
+
+    // if keydown event is not on heading element, ignore
+    if (!heading) return;
+
+    // don’t handle modifier shortcuts used by assistive technology.
+    if (event.altKey) return;
+
+    let newHeading;
```

Comment:
Give this type `HTMLElement | undefined`, just to avoid it having any.

I thought the linter would pick this up since it's part of strict mode, but it isn't. @surma